pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -180,3 +184,85 @@ type HTTPSDConfig struct {
 	// +optional
 	TLSConfig *v1.SafeTLSConfig `json:"tlsConfig,omitempty"`
 }
+
+// HTTPSDConfig defines a prometheus HTTP service discovery configuration
+// See https://prometheus.io/docs/prometheus/latest/configuration/configuration/#consul_sd_config
+// +k8s:openapi-gen=true
+type ConsulSDConfig struct {
+	// A valid string consisting of a hostname or IP followed by an optional port number.
+	// +kubebuilder:validation:MinLength:=1
+	// +required
+	Server string `json:"server,omitempty"`
+	// Consul ACL Token, If not provided it will use the ACL from the local Consul Agent.
+	// +optional
+	Token *corev1.SecretKeySelector `json:"token,omitempty"`
+	// Consul Datacenter name, If not provided it will use the local Consul Agent Datacenter.
+	// +optional
+	Datacenter *string `json:"datacenter,omitempty"`
+	// Namespaces are only supported in Consul Enterprise.
+	// +optional
+	Namespace *string `json:"namespace,omitempty"`
+	// Admin Partitions are only supported in Consul Enterprise.
+	// +optional
+	Partition *string `json:"partition,omitempty"`
+	// HTTP Scheme default "http"
+	// +optional
+	Scheme *string `json:"scheme,omitempty"`
+	// A list of services for which targets are retrieved. If omitted, all services are scraped.
+	// +optional
+	Services []string `json:"services,omitempty"`
+	// An optional list of tags used to filter nodes for a given service. Services must contain all tags in the list.
+	// +optional
+	Tags []string `json:"tags,omitempty"`
+	// The string by which Consul tags are joined into the tag label.
+	// default ","
```

Comment:
(nit) I wouldn't mention the Prometheus default value as it might be confusing and potentially out of sync Maybe this instead?

```suggestion
	// If unset, Prometheus uses its default value.
```